python/ray/data/_internal/logical/util.py

```diff
@@ -70,7 +70,7 @@ def _collect_operators_to_dict(op: LogicalOperator, ops_dict: Dict[str, int]):
     op_name = op.name
 
     # Check read and write operator, and anonymize user-defined data source.
-    if isinstance(op, Read) or isinstance(op, Write):
+    if isinstance(op, Read):
```

Comment:
The unit test to catch this bug is added in https://github.com/ray-project/ray/pull/33168.